OpenRA.Game/Sound/Sound.cs

```diff
@@ -204,11 +206,17 @@ public void PlayMusicThen(MusicInfo m, Action then)
 
 			StopMusic();
 
-			musicSource = getSoundSource(m.Filename);
-			if (musicSource == null)
+			Func<ISoundFormat, ISound> stream = soundFormat => soundEngine.Play2DStream(
+				soundFormat.GetPCMInputStream(), soundFormat.Channels, soundFormat.SampleBits, soundFormat.SampleRate,
+				false, true, WPos.Zero, MusicVolume);
+			music = LoadSound(m.Filename, stream);
+
+			if (music == null)
+			{
+				onMusicComplete = null;
```

Comment:
The current behaviour is to ignore the callback if music fails to load. I'm just taking the opportunity to also null out the callback since it will never get called. The callback is only invoked is music is playing and reaches completion.

The current uses are to switch to the next track automatically when the current one ends, and also as a callback from Lua (which I assume is for much the same purpose).

If we did want to change this, we'd want to make sure it didn't invoke infinite loops from failure to play things.